src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/AwsSdk/KinesisHelper.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NewRelic.Agent.Extensions.AwsSdk
+{
+    public static class KinesisHelper
+    {
+
+        private static readonly ConcurrentDictionary<string, string> _streamNameCache = new();
+        private static readonly ConcurrentDictionary<Type, List<string>> _propertyInfoCache = new();
+
+        public static string GetStreamNameFromRequest(dynamic request)
```

Comment:
Is the request type always public? Is it possible that it's a Request<T>, or Request<Kinesis<T>, where T is not a public type? If it's potentially not public, we can't use `dynamic`.

Reply:
This comment applies to the other methods too.

Reply:
Refactored to avoid using dynamic entirely.